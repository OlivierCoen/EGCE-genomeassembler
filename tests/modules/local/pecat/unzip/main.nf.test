nextflow_process {

    name "Test Process PECAT_UNZIP"
    script "modules/local/pecat/unzip/main.nf"
    process "PECAT_UNZIP"
    tag "pecat"

    test("Should run without failures") {

        when {

            process {
                """
                meta = [ id: "test", genome_size: 1500000 ]
                reads_file = file( '$projectDir/tests/test_data/pecat/unzip/reads.fasta.gz', checkIfExists: true)
                parameters = file( '$projectDir/tests/test_data/pecat/unzip/pecat_optional_config.txt', checkIfExists: true)
                input[0] = [ meta, reads_file ]
                input[1] = parameters
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Very small fasta") {
        tag "pecat_very_small"

        when {

            process {
                """
                meta = [ id: "test", genome_size: 50000 ]
                reads_file = file( '$projectDir/tests/test_data/pecat/unzip/reads.small.fasta.gz', checkIfExists: true)
                parameters = file( '$projectDir/tests/test_data/pecat/unzip/pecat_optional_config.txt', checkIfExists: true)
                input[0] = [ meta, reads_file ]
                input[1] = parameters
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
