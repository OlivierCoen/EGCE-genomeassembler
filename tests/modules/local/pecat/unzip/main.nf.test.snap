{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "10": [
                    
                ],
                "11": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "corrected_reads": [
                    
                ],
                "filtered_overlaps": [
                    
                ],
                "overlaps": [
                    
                ],
                "phased_alternate_assembly": [
                    
                ],
                "phased_primary_assembly": [
                    
                ],
                "phased_rest_assembly": [
                    
                ],
                "polished_alternate_assembly": [
                    
                ],
                "polished_primary_assembly": [
                    
                ],
                "prepared_reads": [
                    
                ],
                "unphased_alternate_assembly": [
                    
                ],
                "unphased_primary_assembly": [
                    
                ],
                "unphased_rest_assembly": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-05-16T17:56:18.099651836"
    },
    "Very small fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "primary.fasta:md5,d8f2d82bbbccfd6aa03f4c85b9719b53"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "alternate.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_1.fasta:md5,80325eb093f164c0dcc8cd193f4c6149"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_2.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "primary.fasta:md5,8807200270eee6ebb807e68792d3054f"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "alternate.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_1.fasta:md5,91b81beb1c9bef2fb5193ffa4760f3ca"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_2.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "polished_medaka_alternate_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "alternate.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "polished_medaka_haplotype_1_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_1.fasta:md5,80325eb093f164c0dcc8cd193f4c6149"
                    ]
                ],
                "polished_medaka_haplotype_2_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_2.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "polished_medaka_primary_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "primary.fasta:md5,d8f2d82bbbccfd6aa03f4c85b9719b53"
                    ]
                ],
                "polished_racon_alternate_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "alternate.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "polished_racon_haplotype_1_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_1.fasta:md5,91b81beb1c9bef2fb5193ffa4760f3ca"
                    ]
                ],
                "polished_racon_haplotype_2_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "haplotype_2.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "polished_racon_primary_assembly": [
                    [
                        {
                            "id": "test",
                            "genome_size": 50000
                        },
                        "primary.fasta:md5,8807200270eee6ebb807e68792d3054f"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-20T09:39:07.952041769"
    }
}